klogr/klogr.go

```diff
@@ -59,6 +59,30 @@ func framesToCaller() int {
 	return 1 // something went wrong, this is safe
 }
 
+func trimDuplicates(kvLists ...[]interface{}) [][]interface{} {
```

Comment:
this function is hardcore. :)

Reply:
This needs a comment to explain how it works as it sort of folds the duplicates left, so the last items in the list take precence.

Do you think it’s worth unit testing this function individually too?

Reply:
yeah, i think adding a few comments to the body would help a lot explaining how it works.
for someone like me with only a few years of golang it's hard.

in terms of unit testing i think `TestInfo` is sufficient.

Reply:
Added a load of additional comments 😄